### ISSUE TITLE
Event cancellation doesn't work right

### ISSUE THREAD

**Issue by EitanBlumin** [MadeiraData/sql-clr-ics#11]
When cancelling an event, I get in Outlook 365 a "not supported calendar message.ics" attachment, which cannot be opened in order to remove the event from the calendar.

**Comment by EitanBlumin** [comment]
I verified that the ICS contents are valid using a couple of online validators, and it passed validation.

I tried all the methods mentioned here:
https://stackoverflow.com/questions/27662899/outlook-365-receives-a-message-that-has-an-attachment-that-is-named-not-support

But nothing helped.

Currently, my suspicion is that this happened because I used a SendGrid account to send the event cancellation, which may have been detected as a sender other than the organizer.

I should test this on a new event created directly from a non-SendGrid account.

**Comment by EitanBlumin** [comment]
The same problem still occurs even when sending from a Gmail account, and when sending from a personal Microsoft account.

I'm currently at a loss. I have no idea why this is happening.

**Comment by bhodge34** [comment]
Not sure if you are still supporting this, but I found that changing the name of the ics file from Meeting.ics to invite.ics caused the cancel mail to work.

**Comment by EitanBlumin** [comment]
Wait, seriously? That easy? I'm gonna try this myself ASAP

**Comment by EitanBlumin** [comment]
> Not sure if you are still supporting this, but I found that changing the name of the ics file from Meeting.ics to invite.ics caused the cancel mail to work.

I tried making the change but I'm still getting the same issue with the "not supported calendar message.ics" attachment.

I pushed the changes to the [v2.4 branch](https://github.com/MadeiraData/sql-clr-ics/tree/v2.4). Are you sure about this solution?

**Comment by bhodge34** [comment]
You are correct.  I was getting it to work when I was sending the invite attachment from gmail.  If I send it from the SQL server to my outlook account, it is giving the same error.  I will keep looking into it.

**Comment by bhodge34** [comment]
I spent a few more hours looking into this, and am unable to figure out why it isn't working.  The work around that I came up with was to send an update meeting request in the distant past and set a mailbox rule to auto delete events older than x number of days.  I know this won't work for everyone, but it may help some.

**Comment by EitanBlumin** [comment]
For reference, I generated a calendar invite using the native Outlook web-based interface and then cancelled it.
I captured the attached **invite.ics** file that arrived with the cancellation, and I'm adding it here:
[invite.zip](https://github.com/user-attachments/files/16640636/invite.zip)

I don't have the time to do it right now, but I should compare this file's contents to the contents generated by my CLR method, and take note of the differences. Maybe I'll find something.

**Comment by Julien22bzh** [comment]
Hi Eitan, 
I hope you're doing well!
I'm taking advantage of the holidays to work on improving my invitation script. Have you had a chance to compare the invitation file sent with the native Outlook Web one?
Best regards
Ju

**Comment by Julien22bzh** [comment]
After a little analysis, I noticed that in the initial invitation and in an update, the ATTENDEE and ORGANIZER fields do not change. However, for a cancel, I see that these fields are modified. Wouldn't that be a clue?

Ju
